lua/ttt2/libraries/marker_vision.lua

```diff
@@ -33,10 +35,13 @@ markerVision.registry = {}
 -- @realm shared
 function markerVision.RegisterEntity(ent, owner, visibleFor, color, receiverList, passThroughData)
 	if SERVER then
-		local plysTeam = GetTeamFilter(owner:GetTeam(), true, true)
+		-- handle the player being in a team that is not known, like the unknown
+		if IsPlayer(owner) then
+			local plysTeam = GetTeamFilter(owner:GetTeam(), false, true)
 
-		if #plysTeam == 0 then -- happens for TEAM_NONE for example
-			visibleFor = VISIBLE_FOR_PLAYER
+			if #plysTeam == 0 then -- happens for TEAM_NONE for example
+				visibleFor = VISIBLE_FOR_PLAYER
+			end
```

Comment:
whoops, seems like i missed this in the last review...

If I understand correctly this means that in this case the MarkerEntity would not carry over to a new role or team if the player owner would change roles/teams, right? This is probably irrelevant for TEAM_NONE but if other teams would suffer from this i'd like to fix this somehow.

Suppose the dev of the melon mine intends for the team of the player to be informed about the location of the mine. They would set `VISIBLE_FOR_TEAM`, but maybe the player is currently in a team that suffers from this condition and then gets its `visibleFor` value overridden. The player then changes team, but the team does not get informed about the markerEntity as it is now only `VISIBLE_FOR_PLAYER`.
Did i understand this correctly?

Reply:
I think this only can happen for players in team none, because this is the only team that set the `.alone` flag. And I don't know to which team I should attach it here as the player has "team none"

Reply:
If I'd attach it to team none a unknow could see the marks of an amnesiac

Reply:
Couldn't we stuff the handling for this case into the if-clause on line 57 then? that way we don't mangle the intention of the consumer.

Reply:
so you mean it will still get changed to "visible only to you" internally, gut it is registered as whatever it was before so that it can carry over on a team change?

Reply:
yes exactly

Reply:
Sounds doable, let me try